pkg/platform/runtime/setup/setup.go

```diff
@@ -401,19 +404,47 @@ func (s *Setup) fetchAndInstallArtifactsFromBuildPlan(installFunc artifactInstal
 
 	// Compute and handle the change summary
 	// runtimeAndBuildtimeArtifacts records all artifacts that will need to be built in order to obtain the runtime.
-	// Disabled due to DX-2033.
-	// Please use this var when we come back to this in the future as we need to make a clear distinction between this
-	// and runtime-only artifacts.
-	// var runtimeAndBuildtimeArtifacts artifact.Map
+	var runtimeAndBuildtimeArtifacts artifact.Map
 	var runtimeArtifacts artifact.Map // Artifacts required for the runtime to function
-	if buildResult.Build != nil {
-		runtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build)
+
+	// If the build is not ready, we need to get the runtime and buildtime closure
+	if !buildResult.BuildReady {
+		runtimeAndBuildtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build, true)
+		if err != nil {
+			return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
+		}
+	}
+
+	var includeBuildtimeClosure bool
+	// If we are installing build dependencies, then buildtime dependencies are also runtime dependencies
+	if strings.EqualFold(os.Getenv(constants.InstallBuildDependencies), "true") {
+		logging.Debug("Installing build dependencies")
+		includeBuildtimeClosure = true
+		if runtimeAndBuildtimeArtifacts == nil {
+			runtimeAndBuildtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build, includeBuildtimeClosure)
+			if err != nil {
+				return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
+			}
+		}
+		runtimeArtifacts = runtimeAndBuildtimeArtifacts
+	} else {
+		runtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build, false)
 		if err != nil {
 			return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
 		}
 	}
 
-	setup, err := s.selectSetupImplementation(buildResult.BuildEngine, runtimeArtifacts)
+	var resolver ArtifactResolver
+	if !buildResult.BuildReady {
+		resolver, err = selectArtifactResolver(buildResult.BuildEngine, runtimeAndBuildtimeArtifacts)
+	} else {
+		resolver, err = selectArtifactResolver(buildResult.BuildEngine, runtimeArtifacts)
```

Comment:
This seems to be handling implicit logic. I'd prefer we rewrite/rename things to make it explicit, or if that's not possible, add a comment.